Adafruit_Si7021.cpp

```diff
@@ -32,76 +28,134 @@
 Adafruit_Si7021::Adafruit_Si7021(void) {
   _i2caddr = SI7021_DEFAULT_ADDRESS;
   sernum_a = sernum_b = 0;
+  _model = "Si7021";
 }
 
 bool Adafruit_Si7021::begin(void) {
   Wire.begin();
 
   reset();
-  if (readRegister8(SI7021_READRHT_REG_CMD) != 0x3A) return false;
+  if (_readRegister8(SI7021_READRHT_REG_CMD) != 0x3A)
+		return false;
 
   readSerialNumber();
-
-  //Serial.println(sernum_a, HEX);
-  //Serial.println(sernum_b, HEX);
+  _readRevision();
 
   return true;
 }
 
 float Adafruit_Si7021::readHumidity(void) {
   Wire.beginTransmission(_i2caddr);
-  Wire.write((uint8_t)SI7021_MEASRH_NOHOLD_CMD);
-  Wire.endTransmission(false);
-  delay(25);
-
-  Wire.requestFrom(_i2caddr, 3);
-  uint16_t hum = Wire.read();
-  hum <<= 8;
-  hum |= Wire.read();
-  uint8_t chxsum = Wire.read();
-
-  float humidity = hum;
-  humidity *= 125;
-  humidity /= 65536;
-  humidity -= 6;
-
-  return humidity;
+  Wire.write(SI7021_MEASRH_NOHOLD_CMD);
+  uint8_t err = Wire.endTransmission(false);
+#ifdef ARDUINO_ARCH_ESP32
+  if(err != I2C_ERROR_CONTINUE) //ESP32 has to queue ReSTART operations.
+#else
+  if (err != 0)
+#endif
+		return NAN; //error
+
+  uint32_t start = millis(); // start timeout
+  while(millis()-start < _TRANSACTION_TIMEOUT){
+	  if (Wire.requestFrom(_i2caddr, 3) == 3) {
+		uint16_t hum = Wire.read() << 8 | Wire.read();
+		uint8_t chxsum = Wire.read();
+
+		float humidity = hum;
+		humidity *= 125;
+		humidity /= 65536;
+		humidity -= 6;
+
+		return humidity;
+      }
+	  delay(6); // 1/2 typical sample processing time
+  }
+  return NAN; // Error timeout
 }
 
 float Adafruit_Si7021::readTemperature(void) {
   Wire.beginTransmission(_i2caddr);
-  Wire.write((uint8_t)SI7021_MEASTEMP_NOHOLD_CMD);
-  Wire.endTransmission(false);
-  delay(25);
-
-  Wire.requestFrom(_i2caddr, 3);
-  uint16_t temp = Wire.read();
-  temp <<= 8;
-  temp |= Wire.read();
-  uint8_t chxsum = Wire.read();
-
-  float temperature = temp;
-  temperature *= 175.72;
-  temperature /= 65536;
-  temperature -= 46.85;
+  Wire.write(SI7021_MEASTEMP_NOHOLD_CMD);
+  uint8_t err = Wire.endTransmission(false);
 
-  return temperature;
+#ifdef ARDUINO_ARCH_ESP32
+  if (err != I2C_ERROR_CONTINUE) //ESP32 has to queue ReSTART operations.
+#else
+  if(err != 0)
+#endif
+  	return NAN; //error
+    
+  uint32_t start = millis(); // start timeout
+  while(millis()-start < _TRANSACTION_TIMEOUT){
+	  if (Wire.requestFrom(_i2caddr, 3) == 3) {
+  		uint16_t temp = Wire.read() << 8 | Wire.read();
+  		uint8_t chxsum = Wire.read();
```

Comment:
space instead of tab
add a comment that chxsum is not used and could be a `todo: implement checksum`